### ISSUE TITLE
Enhance configurability of estimators

### ISSUE THREAD

**Issue by sekruse** [daqcri/rheem#1]
Not all `Operator`s currently support proper configurability of estimators for UDFs. Moreover, the data distribution is unknown to Rheem, while the user might profit useful hints here and there. This pertains to two metrics:

1. Cardinality estimation
  * influenced by UDFs (at least): `FilterOperator`, `FlatMapOperator`
  * influenced by data distribution (at least): `DistinctOperator`, `FilterOperator`, `FlatMapOperator` (via broadcasts), `MaterializedGroupByOperator`, `IntersectOperator`, `JoinOperator`, `ReduceByOperator`
2. Load estimation
  * basically, every `ExecutionOperator` with a UDF might want to consider the load of its UDF - in particular for heavy-weight UDFs

These observations yield the following tasks:

- [x] allow to configure the selectivity of UDFs for `FilterOperator`s and `FlatMapOperator`s
- [x] allow to override the `CardinalityEstimator`s for `Operator` instances
- [x] allow to configure `LoadProfileEstimator`s for UDFs
- [x] take into account UDF `LoadProfileEstimator`s in the respective `ExecutionOperator`s' `LoadProfileEstimator`s

**Comment by luckyasser** [comment]
This issue was moved to rheem-ecosystem/rheem#11
